packages/contracts/deploy/018-fund-accounts.ts

```diff
@@ -0,0 +1,66 @@
+/* Imports: External */
+import { sleep } from '@eth-optimism/core-utils'
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import {
+  defaultHardhatNetworkHdAccountsConfigParams,
+  defaultHardhatNetworkParams,
+} from 'hardhat/internal/core/config/default-config'
+import { normalizeHardhatNetworkAccountsConfig } from 'hardhat/internal/core/providers/util'
+
+/* Imports: Internal */
+import { getDeployedContract } from '../src/hardhat-deploy-ethers'
+
+// This is a TEMPORARY way to fund the default hardhat accounts on L2. The better way to do this is
+// to make a modification to hardhat-ovm. However, I don't have the time right now to figure the
+// details of how to make that work cleanly. This is fine in the meantime.
+const deployFn: DeployFunction = async (hre) => {
+  // Only execute this step if we're on the hardhat chain ID.
+  const { chainId } = await hre.ethers.provider.getNetwork()
```

Comment:
Since this is destructuring here, its definitely returning an object with `chainId` in it but i'm wondering why not call `provider.chainId`? In the case where the network id can not match the chain id

Reply:
I don't think there is a `provider.chainId`

Reply:
You are right, i was mistaking `wallet.getChainId` which wraps `provider.getNetwork`

https://github.com/ethers-io/ethers.js/blob/4e6d121fb8aa7327290afab7653364be8ddd8d81/packages/abstract-signer/src.ts/index.ts#L131